src/main/java/com/example/viacademy/entities/User.java

```diff
@@ -0,0 +1,45 @@
+package com.example.viacademy.entities;
+
+import com.example.viacademy.entities.pivots.CourseUser;
+import jakarta.persistence.*;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Date;
+import java.util.List;
+
+@Entity
+@Table(name = "users")
+@Getter
+@Setter
+public class User {
```

Comment:
in this table only keep the information user (ex email, name, etc). create another table and name it as `InstructorProfile`, that table will be used if the user will become an instructor, and in that table you'll save data about courses, payment, etc. in the case the user will buy courses, you can create a table `StudentProfile` and in that table you'll save information relate it